File: utils/src/main/java/ru/virgil/utils/image/IPrivateImageRepository.java
package ru.virgil.utils.image;

import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.NoRepositoryBean;

import java.util.Optional;
import java.util.UUID;

@NoRepositoryBean
public interface IPrivateImageRepository<Owner extends IBaseEntity, PrivateImage extends IPrivateImage<Owner>>
        extends CrudRepository<PrivateImage, UUID> {

    Optional<PrivateImage> findByOwnerAndUuid(Owner owner, UUID imageUuid);

}

File: utils/src/main/java/ru/virgil/utils/image/IPrivateImageRepository.kt
package ru.virgil.utils.image

import org.springframework.data.repository.CrudRepository
import org.springframework.data.repository.NoRepositoryBean
import java.util.*

@NoRepositoryBean
interface IPrivateImageRepository<Owner : IBaseEntity, PrivateImage : IPrivateImage<Owner>> :
    CrudRepository<PrivateImage, UUID> {

    fun findByOwnerAndUuid(owner: Owner, imageUuid: UUID): Optional<PrivateImage>
}
